src/tests/data/restore.test.ts

```diff
@@ -0,0 +1,703 @@
+import * as restore from "../../data/restore";
+import {
+  ExcalidrawTextElement,
+  ExcalidrawFreeDrawElement,
+  ExcalidrawLinearElement,
+  ExcalidrawElement,
+} from "../../element/types";
+import * as sizeHelpers from "../../element/sizeHelpers";
+import { API } from "../helpers/api";
+import { getDefaultAppState } from "../../appState";
+import { ImportedDataState } from "../../data/types";
+import { NormalizedZoomValue } from "../../types";
+import {
+  FONT_FAMILY,
+  DEFAULT_FONT_FAMILY,
+  DEFAULT_TEXT_ALIGN,
+  DEFAULT_VERTICAL_ALIGN,
+} from "../../constants";
+
+const mockSizeHelper = jest.spyOn(sizeHelpers, "isInvisiblySmallElement");
+
+beforeEach(() => {
+  mockSizeHelper.mockReset();
+});
+
+describe("restoreElements", () => {
+  it("with null array of elements", () => {
+    expect(restore.restoreElements(null)).toStrictEqual([]);
+  });
+
+  it("should not call isInvisiblySmallElement when input element is a selection element", () => {
+    const selectionEl = { type: "selection" } as ExcalidrawElement;
+    const restoreElements = restore.restoreElements([selectionEl]);
+    expect(restoreElements.length).toBe(0);
+    expect(sizeHelpers.isInvisiblySmallElement).toBeCalledTimes(0);
+  });
+
+  it("when input element type is not supported", () => {
+    const dummyNotSupportedElement = API.createElement({ type: "text" });
+
+    Object.defineProperty(dummyNotSupportedElement, "type", {
+      get: jest.fn(() => "not supported element type"),
+    });
+
+    expect(restore.restoreElements([dummyNotSupportedElement]).length).toBe(0);
+  });
+
+  it("when isInvisiblySmallElement is true", () => {
+    const rectElement = API.createElement({ type: "rectangle" });
+    mockSizeHelper.mockImplementation(() => true);
+
+    expect(restore.restoreElements([rectElement]).length).toBe(0);
+  });
+
+  it("with text element type", () => {
+    const textElement = API.createElement({
+      type: "text",
+      fontSize: 14,
+      fontFamily: 1,
+      text: "text",
+      textAlign: "center",
+      verticalAlign: "middle",
```

Comment:
```suggestion
```

let's remove this so restoring correctly is asserted

Reply:
as we talked before, I've separated the tests for text elements in two: 1) passing all attributes 2) not passing all attributes and letting the restore function do its job :)

Reply:
@ad1992 I updated the tests with the snapshots :)